src/docket/dependencies.py

```diff
@@ -55,6 +56,26 @@ def TaskKey() -> str:
     return cast(str, _TaskKey())
 
 
+class _TaskLogger(Dependency):
+    def __call__(
+        self, docket: Docket, worker: Worker, execution: Execution
+    ) -> logging.LoggerAdapter:
+        logger = logging.getLogger(f"docket.task.{execution.key}")
```

Comment:
We could make this configurable but I kind of feel like we always want the logger to be the task identifier?